test_spack.py

```diff
@@ -0,0 +1,214 @@
+    'cosmo' : {
+        'spack installcosmo cosmo@master%pgi cosmo_target=gpu +cppdycore', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+        'spack installcosmo cosmo@master%pgi cosmo_target=cpu ~cppdycore', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+        'git clone git@github.com:MeteoSwiss-APN/cosmo.git \
+            && cd cosmo \
+            && spack devbuildcosmo cosmo@dev-build%pgi cosmo_target=gpu +cppdycore \
+            && cd .. && rm -rf cosmo', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html 
+        'git clone git@github.com:MeteoSwiss-APN/cosmo.git \
+            && cd cosmo \
+            && spack devbuildcosmo cosmo@dev-build%pgi cosmo_target=cpu ~cppdycore \
+            && cd .. && rm -rf cosmo', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+        # 'spack installcosmo cosmo@master%pgi cosmo_target=gpu +cppdycore ^eccodes +aec +build_shared_libs',
+    },
+    'cosmo-dycore' : {},
+    'cosmo-eccodes-definitions' : {},
+    'cosmo-grib-api' : {},
+    'cosmo-grib-api-definitions' : {},
+    'cuda' : {},
+    'dawn' : {},
+    'dawn4py' : {},
+    'dusk' : {},
+    'dyicon_benchmarks' : {},
+    'ecbuild' : {},
+    'eccodes' : {},
+    'eckit' : {},
+    'gridtools' : {},
+    'icon' : {
+        # 'git clone --recursive git@gitlab.dkrz.de:icon/icon-cscs.git \
+        #     && cd icon-cscs \
+        #     && mkdir pgi_cpu \
+        #     && cd pgi_cpu \
+        #     && touch a_fake_file.f90 \
+        #     && spack dev-build -u build icon@dev-build%pgi config_dir=./.. icon_target=cpu', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+        # 'git clone --recursive git@gitlab.dkrz.de:icon/icon-cscs.git \
+        #     && cd icon-cscs \
+        #     && mkdir pgi_gpu \
+        #     && cd pgi_gpu \
+        #     && touch a_fake_file.f90 \
+        #     && spack dev-build -u build icon@dev-build%pgi config_dir=./.. icon_target=gpu', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+    },
+    'icondusk-e2e' : {},
+    'icontools' : {},
+    'int2lm' : {
+        'spack install int2lm@c2sm_master%pgi', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+        'spack install int2lm@org_master%pgi pollen=False', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+    },
+    'libgrib1' : {},
+    'mpich' : {},
+    'oasis' : {},
+    'omnicompiler' : {},
+    'omni-xmod-pool' : {},
+    'openmpi' : {},
+    'serialbox' : {},
+    'xcodeml-tools' : {},
+    'zlib_ng' : {},
+}
+
+def UseCases(packages : set):
+    return {case for p in packages for case in use_cases[p]}
+
+# Maps packages to Set[packages it depends on].
+dependencies = {
+    'atlas' : {'ecbuild', 'eckit'},
+    'atlas_utilities' : {'atlas', 'eckit'},
+    'claw' : {},
+    'cosmo' : {'cuda', 'serialbox', 'libgrib1', 'cosmo-grib-api-definitions', 'cosmo-eccodes-definitions', 'omni-xmod-pool', 'claw', 'zlib_ng'},
```

Comment:
depending on the build, cosmo will depend on cosmo-dycore

Reply:
Good catch, I forgot to add a ToDo for that.
The ToDo is to find a good solution for a conditional dependency. I came up with these ideas:
1. Adding it as an unconditional dependency would mean, if we trigger cosmo-dycore, all the use cases of cosmo will be triggered, even the ones that don't really depend on cosmo-dycore.
2. If we don't include it as a dependency the burden is on the people that launch the tests. They need to decide if they should only trigger cosmo-dycore or if they need cosmo as well.
3. If we add a mechanism to conditionally include that dependency, we duplicate the logic of the cosmo package. That reduces testing time for maintenance time.

I'd go for option 1 and change to option 3 if testing time is an issue. What do you think?

Reply:
yes, I would also go for 1. and not add contiditionals to the logic. Because too complex otherwise